File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length <= 1){
            return 0;
        }
        
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;

        for (int i =0; i<prices.length; i++){
            if (prices[i] < min)
                min = prices[i];
            
            int max1 = prices[i] - min;
            
            if (max1 > max ){
                max = max1;
            }
        }
        return max;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1:
            return 0
        
        min = float('inf')
        max = float('-inf')
        for i in prices:
            if i < min:
                min = i
            
            if i - min > max:
                max = i - min
                
        return max